### PR TITLE
CTM360-hv-cbs-integration-FunctionApp

### PR DESCRIPTION
**Pull request by edx-sayed-salman**
Required items, please complete

Change(s):

Added new solution and support FunctionApp connector.
Reason for Change(s):

Added new solution and support FunctionApp connector.
Version Updated:

NA
Testing Completed:

Yes
Checked that the validations are passing and have addressed any issues that are present:

NA

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, Thanks for raising this PR. We will investigate this PR and will update you about the same before 14 December, 2023.

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, This PR is still under investigation we will come back to you with an update by 19 December, 2023. 
Until then can you please try to resolve **Code QL** errors.

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, Please create 2 custom tables named as `CBSLog_Azure_1_CL` && `HackerViewLog_Azure_1_CL` at location `.script/tests/KqlvalidationsTests/CustomTables`
You can refer any of the other table from the folder for more clarification.

File: Solutions/CTM360/Data Connectors/CBS/CTM360_CBS_API_functionApp.json [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        },
+        {
+            "title": "2. Import Function App Code(Zip deployment)",
+            "description": "1. Install Azure CLI\n2. From terminal type **az functionapp deployment source config-zip -g <ResourceGroup> -n <FunctionApp> --src <Zip File>** and hit enter. Set the `ResourceGroup` value to: your resource group name. Set the `FunctionApp` value to: your newly created function app name. Set the `Zip File` value to: `CTM360Connector.zip`(path to your zip file). Note:- Download the zip file from the link - [Function App Code](https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/CTM360/Data%20Connectors/CTM360/CTMConnector.zip)"
```
Comment (v-prasadboke) [inline Solutions/CTM360/Data Connectors/CBS/CTM360_CBS_API_functionApp.json:122]: Please correct the instructions, if you want to add cbs function app or ctm360 function app. Also please correct the hyper link which will refer to the raw zip file

File: Solutions/CTM360/Data Connectors/HackerView/CTM360_HV_API_FunctionApp.json [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        },
+        {
+            "title": "2. Import Function App Code(Zip deployment)",
+            "description": "1. Install Azure CLI\n2. From terminal type **az functionapp deployment source config-zip -g <ResourceGroup> -n <FunctionApp> --src <Zip File>** and hit enter. Set the `ResourceGroup` value to: your resource group name. Set the `FunctionApp` value to: your newly created function app name. Set the `Zip File` value to: `CTM360Connector.zip`(path to your zip file). Note:- Download the zip file from the link - [Function App Code](https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/CTM360/Data%20Connectors/CTM360/CTMConnector.zip)"
```
Comment (v-prasadboke) [inline Solutions/CTM360/Data Connectors/HackerView/CTM360_HV_API_FunctionApp.json:122]: Same is the case here

**Comment by edx-sayed-salman** [comment]
Hi @v-prasadboke , please have alook at the changes , also can you check why logo fails? thank you

**Comment by v-prasadboke** [comment]
@edx-sayed-salman I'll take a look at it.

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, I'm trying to resolve the validation error for Logo. Will get back to you soonest as soon as its done

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, try removing the style attribute from logo. 
 Logo needs to be in SVG format and under 5 Kb

**Ensure raw file of logo does not have any of the following:**
1. cls and style formats
2. embedded png formats
3. xmlns:xlink
4. Logo needs to be in SVG format and under 5 Kb
5. Logo scales well to fit in a 75 px square
6. SVG code file is formatted well for readability

**Comment by edx-sayed-salman** [comment]
Hi @v-prasadboke can you have a look, the image is solved , and please check also the latest code. thank you

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, THanks for commiting the changes will investigate it and come back to you by 29 December, 2023.

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, All looks good. Just let me check once Code QL errors and meanwhile can you share working images of function app. Thanks.

**Comment by edx-sayed-salman** [comment]
dear @v-prasadboke here is some working function photos. 
image
image
image
image

**Review by v-prasadboke — DISMISSED** [review]

**Review by v-prasadboke — APPROVED** [review]

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**